src/js/angular/clustermanagement/directives/cluster-list.directive.js

```diff
@@ -0,0 +1,198 @@
+                    .then((location) => {
+                        if (location.error) {
+                            handleErrors(location.error);
+                            ClusterContextService.updateClusterView();
+                            return;
+                        }
+                        if (location) {
+                            ClusterContextService.addLocation($scope.cluster, new Location(location));
+                        }
+                    })
+                    .catch(function (error) {
+                        handleErrors(error.data, error.status);
+                        ClusterContextService.updateClusterView();
+                    })
+                    .finally(() =>{
+                        $scope.editedNodeIndex = undefined;
+                        $scope.setLoader(false);
+                    });
+            };
+
+            /**
+             * Deletes a node from the cluster.
+             * @param {number} index - The index of the node to delete.
+             */
+            $scope.deleteNode = (index) => {
+                // Not implemented
+            };
+
+            /**
+             * Checks if the current cluster configuration is valid.
+             * The configuration is considered valid if there are at least two nodes and no nodes are being edited.
+             * @return {boolean} True if the configuration is valid, false otherwise.
+             */
+            $scope.isClusterConfigurationValid = () => {
+                const isNotInEditMode = $scope.editedNodeIndex === undefined;
+                const validNodes = $scope.cluster.getViewModel().filter((node) => node.endpoint && node.endpoint.length > 0);
+                const hasValidNodes = validNodes.length >= 2;
+                return isNotInEditMode && hasValidNodes;
+            };
+
+            /**
+             * Sets the loader state with a message.
+             * The loader state is shown with a delay to avoid flickering for short operations.
+             * @param {boolean} loader - Whether to show the loader.
+             * @param {string} [message] - The message to show with the loader.
+             */
+            $scope.setLoader = (loader, message) => {
+                $timeout.cancel($scope.loaderTimeout);
+                if (loader) {
+                    $scope.loaderTimeout = $timeout(() => {
+                        $scope.loader = loader;
+                        $scope.loaderMessage = message;
+                    }, 300);
+                } else {
+                    $scope.loader = false;
+                }
+            };
+
+            // =========================
+            // Private functions
+            // =========================
+            const onClusterViewChanged = (cluster) => {
+                $scope.cluster = cluster;
+                $scope.clusterView = cluster.getAttached();
+                $scope.allSuggestions = cluster.getAvailable().map((node) => node.endpoint);
```

Comment:
You can move/hide/ this and some other similar operations in the model itself. For example in this case you can have `cluster.getAvailableNodeEndpoints()`

Reply:
done